stubs/Pygments/pygments/__init__.pyi

```diff
@@ -1,16 +1,16 @@
 from _typeshed import SupportsWrite
 from typing import TypeVar, overload
 
-from pygments.formatter import Formatter
+from pygments.formatter import _Formatter
 
 _T = TypeVar("_T", str, bytes)
 
 def lex(code, lexer): ...
 @overload
-def format(tokens, formatter: Formatter[_T], outfile: SupportsWrite[_T]) -> None: ...
+def format(tokens, formatter: _Formatter[_T], outfile: SupportsWrite[_T]) -> None: ...
```

Comment:
How am I supposed to call this function now? It should accept any `Formatter`, but it does not. I think hiding the generic-ness in a non-public class `_Formatter` only moves the problem.

If you get a runtime error for `Formatter[str]`, you should consider using `from __future__ import annotations`. Many other stubs have the same situation, and we are used to telling this to users. See e.g. #5652.

Reply:
Right good point, I didn't consider that. In that case I think we should revert #6819 because e.g. the following code:

```python
class Foo(pygments.formatter.Formatter):
    pass
```

should not yield errors when using pyright or mypy with strict settings. But currently it does because of `Missing type parameters for generic type "Formatter"`.

Reply:
Yeah, subclassing is quite annoying. One possible workaround is:

```python3
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    _Base = pygments.formatter.Formatter[str]
else:
    _Base = pygments.formatter.Formatter

class Foo(_Base):
    pass
```

but it's a mess. Apart from subclassing, things should just work most of the time, because the type of e.g. `foo = Formatter(encoding="utf-8")` is `Formatter[bytes]`.

Making the actual `Formatter` inherit from `Generic` would probably be the best. In my experience, pygments maintainers merge pull requests eventually, but it usually takes a few weeks.

Reply:
As @Akuli says, the best solution here is probably to get a fix merged upstream with pygments. Longer term, a general solution would probably be something like https://github.com/python/typing/issues/956 or https://github.com/python/typing/issues/307

Reply:
Right I opened a PR for pygments: https://github.com/pygments/pygments/pull/2081. I'll update this PR accordingly so that it only fixes the `ImageFormatter` situation.